### ISSUE TITLE
Investigate better approach to value to string-ish comparison

### ISSUE THREAD

**Issue by grisumbras** [boostorg/json#673]
Currently:
```c++
    friend
    bool
    operator==(
        value const& lhs,
        value const& rhs) noexcept
    {
        return lhs.equal(rhs);
    }
```

Thus, `jv == "some string"` is equivalent to `jv == value("some string")`, which allocates.
Consider adding an overload similar to one for `string` .

**Comment by vinniefalco** [comment]
I'm ok with this though

**Comment by pdimov** [comment]
```
bool op==( value const&, string_view) noexcept;
bool op==( string_view, value const&) noexcept;
```
but this could introduce ambiguities.

**Comment by vinniefalco** [comment]
I don't think we should do this. Instead, we should document the rationale for why we aren't doing it.

**Comment by grisumbras** [comment]
I mis-named the issue. The problem isn't just with strings, it's also present with objects and array.
There are 2 issues:
1. `operator==` will silently allocate when a value is compared against a string, an `array` or an `object`.
2. `operator==` works with `string_view`, but not with `std::string_view`.

**Comment by grisumbras** [comment]
Is there a way to compare against initializer list without explicitly spelling out `std::initializer_list`?
